### PR TITLE
chore(flags): cleaning up some flags stuff

### PR DESCRIPTION
**Pull request by dmarticus**
## Problem

Olly left a bunch of comments on my cohorts PRs, so I implemented them here as to not clutter other PRs too much.

Here's the list of comments I addressed:

- https://github.com/PostHog/posthog/pull/25776#discussion_r1843937515
- https://github.com/PostHog/posthog/pull/25776#discussion_r1843940580
- https://github.com/PostHog/posthog/pull/25776#discussion_r1843942190
- https://github.com/PostHog/posthog/pull/25776#discussion_r1843944260
- https://github.com/PostHog/posthog/pull/25776#discussion_r1843946266
- https://github.com/PostHog/posthog/pull/25776#discussion_r1843948442
- https://github.com/PostHog/posthog/pull/25776#discussion_r1843951512
- https://github.com/PostHog/posthog/pull/25776#discussion_r1843961601
- https://github.com/PostHog/posthog/pull/25776#discussion_r1843963901
- https://github.com/PostHog/posthog/pull/25776#discussion_r1843969985
- https://github.com/PostHog/posthog/pull/25776#discussion_r1843976849
- https://github.com/PostHog/posthog/pull/25776#discussion_r1843981470
- https://github.com/PostHog/posthog/pull/25776#discussion_r1844022073

This shouldn't have any functional changes, and all the existing tests passed.

### CODE REVIEW

File: rust/Cargo.toml [hunk 1]
```diff
@@ -96,4 +96,4 @@ ahash = "0.8.11"
 aws-config = { version = "1.1.7", features = ["behavior-version-latest"] }
 aws-sdk-s3 = "1.58.0"
 mockall = "0.13.0"
-moka = { version = "0.12.8", features = ["sync"] }
+moka = { version = "0.12.8", features = ["sync", "future"] }
```
Comment (dmarticus) [inline rust/Cargo.toml:99]: added this feature to the workspace version of moka because I need it for flags

File: rust/feature-flags/Cargo.toml [hunk 2]
```diff
@@ -40,7 +40,7 @@ common-metrics = { path = "../common/metrics" }
 tower = { workspace = true }
 derive_builder = "0.20.1"
 petgraph = "0.6.5"
-moka = { version = "0.12.8", features = ["future"] }
+moka = { workspace = true }
```
Comment (dmarticus) [inline rust/feature-flags/Cargo.toml:43]: use the workspace version

File: rust/feature-flags/src/config.rs [hunk 3]
```diff
@@ -98,6 +98,12 @@ pub struct Config {
 
     #[envconfig(from = "TEAM_IDS_TO_TRACK", default = "all")]
     pub team_ids_to_track: TeamIdsToTrack,
+
+    #[envconfig(from = "CACHE_MAX_CAPACITY", default = "100000")]
+    pub cache_max_capacity: u64,
```
Comment (havenbarnes) [inline rust/feature-flags/src/config.rs:103]: nit: adding a unit to the name like `cache_ttl_seconds` has would be better? e.g. `cache_max_capacity_bytes` / `cache_max_bytes_capacity`

**Review by havenbarnes — APPROVED** [review]
LG! Just one nit

File: rust/feature-flags/src/config.rs (hunk 3 above)
Comment (dmarticus) [reply rust/feature-flags/src/config.rs:103]: v good point, will do that.

**Pull request merged**